File: fleeman/src/main/java/com/example/demo/entities/Airport.java
package com.example.demo.entities;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class Airport 
{
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
	private int airportId;
	@Column(unique = true)
	private String airportCode;
    private String airportName;
    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "cityId")
    private City cityId;
    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "stateId")
    private State stateId;
    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "hubId")
    private Hub hubId;
    
	public int getAirportId()
	{
		return airportId;
	}

	public String getAirportCode() 
	{
		return airportCode;
	}

	public void setAirportCode(String airportCode)
	{
		this.airportCode = airportCode;
	}

	public String getAirportName()
	{
		return airportName;
	}

	public void setAirportName(String airportName) 
	{
		this.airportName = airportName;
	}

	public City getCityId()
	{
		return cityId;
	}

	public void setCityId(City cityId) 
	{
		this.cityId = cityId;
	}

	public State getStateId()
	{
		return stateId;
	}

	public void setStateId(State stateId) 
	{
		this.stateId = stateId;
	}

	public Hub getHubId()
	{
		return hubId;
	}

	public void setHubId(Hub hubId)
	{
		this.hubId = hubId;
	}
}

File: pro3/Models/Airport.cs
using pro3.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace pro3.Models
{
    public class Airport
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int AirportId { get; set; }

        [Required]
        [MaxLength(255)]
        public string AirportCode { get; set; }

        [Required]
        [MaxLength(255)]
        public string AirportName { get; set; }

        [ForeignKey("CityId")]
        public int? CityId { get; set; }

        [ForeignKey("StateId")]
        public int? StateId { get; set; }

        [ForeignKey("HubId")]
        public int? HubId { get; set; }

        public virtual City? City { get; set; }

        public virtual State? State { get; set; }

        public virtual Hub? Hub { get; set; }
    }
}
